include/common_robotics_utilities/ros_conversions.hpp

```diff
@@ -4,96 +4,118 @@
 
 #include <Eigen/Geometry>
 #include <common_robotics_utilities/math.hpp>
+
+#if COMMON_ROBOTICS_UTILITIES__SUPPORTED_ROS_VERSION == 2
+#include <geometry_msgs/msg/point_stamped.hpp>
+#include <geometry_msgs/msg/pose_stamped.hpp>
+#include <geometry_msgs/msg/transform_stamped.hpp>
+#elif COMMON_ROBOTICS_UTILITIES__SUPPORTED_ROS_VERSION == 1
 #include <geometry_msgs/PointStamped.h>
 #include <geometry_msgs/PoseStamped.h>
 #include <geometry_msgs/TransformStamped.h>
+#else
+#error "Undefined or unknown COMMON_ROBOTICS_UTILITIES__SUPPORTED_ROS_VERSION"
+#endif
 
 namespace common_robotics_utilities
 {
 namespace ros_conversions
 {
-Eigen::Vector3d GeometryPointToEigenVector3d(
-    const geometry_msgs::Point& point);
-
-geometry_msgs::Point EigenVector3dToGeometryPoint(
-    const Eigen::Vector3d& point);
-
-Eigen::Vector4d GeometryPointToEigenVector4d(
-    const geometry_msgs::Point& point);
 
-geometry_msgs::Point EigenVector4dToGeometryPoint(
-    const Eigen::Vector4d& point);
-
-geometry_msgs::PointStamped EigenVector3dToGeometryPointStamped(
+#if COMMON_ROBOTICS_UTILITIES__SUPPORTED_ROS_VERSION == 2
```

Comment:
Is there a reason for the `include` block the be of the form `#if / #elif / #else / #endif` while the `using` block is `#if / #endif / #if / #endif`?
I'd prefer if they both used the `#if / #elif / #else / #endif` form.

Reply:
> Is there a reason for the include block the be of the form #if / #elif / #else / #endif while the using block is #if / #endif / #if / #endif?

It didn't make sense to me to report the missing identifier twice i.e. to have the first `#else` clause in two places. Otherwise, it's the same. We can go with the `#if` / `#elif` / `#endif` form too.

Reply:
See https://github.com/calderpg/common_robotics_utilities/pull/6/commits/0d5cf5f5b4d2b64e413adce4c3b241df447daccc.